### ISSUE TITLE
TodoStore.getAll() Is Always Returning New Object

### ISSUE THREAD

**Issue by charliegeiger89** [MandarinConLaBarba/flux-immutable-todomvc#3]
**TLDR:** The TodoStore is always returning a new object on the `TodoStore.getAll()` method.

---

Awesome example of Flux + Immutable.js, but I stumbled upon an issue when using it as a guideline:

```
  getAll: function() {
    return _todos.toObject();
  }
```

Calling `toObject()` prevents `shouldComponentUpdate` from being able to properly check the diff since it is never receiving the same object. The purpose of using immutable stores is to allow efficient shallow  comparisons **and** more expensive deep comparisons when needed. Returning the raw `_todos` would fix the issue and also allow for `toJS()` and `toObject()` conversion in the component rather than the payload.

[TodoStore Method](https://github.com/MandarinConLaBarba/flux-immutable-todomvc/blob/f75a4f476de2df0889feed9d1a684dbb206b77e2/js/stores/TodoStore.js#L125)

**Comment by MandarinConLaBarba** [comment]
Thanks for the feedback :)

I believe `toObject()` returns a 'shallow' conversion, so the `_todos` (an `OrderedMap`) values are `Record` instances. I believe this is working properly, as I've added the `PureRenderMixin` for the `TodoItem` component and it is indeed only being rendered when its designated `TodoRecord` is mutated.

**Comment by charliegeiger89** [comment]
I did some further testing and it looks like what you think is happening is not actually happening. You can test this by passing `TodoActions.updateText()` the exact same data twice in a row. Immutable `OrderedMap.set()` will return the same object since nothing has updated, but when you call `toObject()` it returns a new object. I will try to set up a gist to demonstrate what I mean.

**Comment by MandarinConLaBarba** [comment]
Yeah I definitely agree that `toObject()` returns a new object, but its values (`Record` instances) are references, and not new/mutated. I realize it would be more advantageous to return immutable objects in components, but I didn't want to muddy the waters, since there are [other write-ups](https://github.com/facebook/immutable-js/wiki/Immutable-as-React-state) on that.

**Comment by prakhar1989** [comment]
Came here to point out this very issue. I agree with @charliegeiger89 that it's better to convert the Immutable collections to primitives in the component rather than the payload. Thank you for sharing this example, btw!
